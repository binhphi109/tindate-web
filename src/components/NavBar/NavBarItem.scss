.navbarItem {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;

  i {
    width: 24px;
  }

  .navbarTitle {
    font-weight: bold;
    color: #fe5858;
  }
}
