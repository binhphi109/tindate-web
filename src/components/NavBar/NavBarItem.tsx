import React, { MouseEventHandler } from "react";
import { IconButton } from "@material-ui/core";
import clsx from "clsx";
import "./NavBarItem.scss";

type Props = {
  className?: string;
  icon?: string;
  iconColor?: string;
  title?: string;
  color?: "inherit" | "primary" | "secondary" | "default";
  onClick?: MouseEventHandler;
};

function NavBarItem({ className, icon, iconColor, title, color, onClick }: Props) {
  return (
    <div className={clsx("navbarItem", className)} onClick={onClick}>
      {icon && (
        <IconButton size="medium" color={color}>
          <i className={icon} style={{ color: iconColor }} />
        </IconButton>
      )}
      {title && <div className="navbarTitle">{title}</div>}
    </div>
  );
}

NavBarItem.defaultProps = {
  className: null,
  icon: null,
  iconColor: null,
  title: null,
  color: "default",
  onClick: null,
};

export default NavBarItem;
