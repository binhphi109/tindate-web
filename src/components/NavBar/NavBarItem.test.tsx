import React from "react";
import { shallow } from "enzyme";
import NavBarItem from "./NavBarItem";

describe("components/navbar/NavBarItem", () => {
  test("should match snapshot with icon", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<NavBarItem icon="fas fa-tasks" onClick={onClick} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot with title", () => {
    const wrapper = shallow(<NavBarItem title="Likes" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should match snapshot with color", () => {
    const wrapper = shallow(<NavBarItem icon="fas fa-fire" color="primary" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("should handle onClick", () => {
    const onClick = jest.fn();
    const wrapper = shallow(<NavBarItem icon="fas fa-chevron-left" onClick={onClick} />);

    wrapper.find("div.navbarItem").simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
