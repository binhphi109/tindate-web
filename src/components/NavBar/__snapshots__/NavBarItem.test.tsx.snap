// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/navbar/NavBar should match snapshot with icon 1`] = `
<div
  className="navbar"
>
  <NavBarItem
    className={null}
    color="default"
    icon="fas fa-tasks"
    iconColor={null}
    onClick={[MockFunction]}
    title={null}
  />
  <NavBarItem
    className={null}
    color="primary"
    icon="fas fa-fire"
    iconColor={null}
    onClick={null}
    title={null}
  />
  <NavBarItem
    className={null}
    color="default"
    icon="far fa-comment"
    iconColor={null}
    onClick={[MockFunction]}
    title={null}
  />
</div>
`;

exports[`components/navbar/NavBar should match snapshot with title 1`] = `
<div
  className="navbar"
>
  <NavBarItem
    className={null}
    color="default"
    icon="fas fa-chevron-left"
    iconColor={null}
    onClick={[MockFunction]}
    title={null}
  />
  <NavBarItem
    className={null}
    color="default"
    icon={null}
    iconColor={null}
    onClick={null}
    title="Likes"
  />
  <NavBarItem
    className={null}
    color="default"
    icon={null}
    iconColor={null}
    onClick={null}
    title={null}
  />
</div>
`;

exports[`components/navbar/NavBarItem should match snapshot with color 1`] = `
<div
  className="navbarItem"
  onClick={null}
>
  <WithStyles(ForwardRef(IconButton))
    color="primary"
    size="medium"
  >
    <i
      className="fas fa-fire"
      style={
        Object {
          "color": null,
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`components/navbar/NavBarItem should match snapshot with icon 1`] = `
<div
  className="navbarItem"
  onClick={[MockFunction]}
>
  <WithStyles(ForwardRef(IconButton))
    color="default"
    size="medium"
  >
    <i
      className="fas fa-tasks"
      style={
        Object {
          "color": null,
        }
      }
    />
  </WithStyles(ForwardRef(IconButton))>
</div>
`;

exports[`components/navbar/NavBarItem should match snapshot with title 1`] = `
<div
  className="navbarItem"
  onClick={null}
>
  <div
    className="navbarTitle"
  >
    Likes
  </div>
</div>
`;
